corehq/form_processor/models.py

```diff
@@ -352,6 +352,16 @@ def metadata(self):
         if const.TAG_META in self.form_data:
             return XFormPhoneMetadata.wrap(clean_metadata(self.form_data[const.TAG_META]))
 
+    def set_meta_properties(self):
+        # doesn't save the form
+        from corehq.apps.receiverwrapper.util import get_app_version_info
+        self.time_end = getattr(self.metadata, 'timeEnd', None)
+        self.time_start = getattr(self.metadata, 'timeStart', None)
+        app_version_info = get_app_version_info(self.domain, self.build_id, None, self.metadata)
```

Comment:
This could return `None`

Reply:
It doesn't look like `get_app_version_info ` will ever return `None`, does it?

Reply:
sorry was thinking of `get_meta_appversion_text`